File: apps/ios/SofaExchange/Domain/Entities/Listing.swift
import Foundation

struct Listing {
    let id: String
    let title: String
    let city: City
    let pricePerNightCents: Int
    let sofaType: SofaType
    let hasFreeWifi: Bool
    let createdAt: String
    let updatedAt: String

    var priceDisplay: String {
        String(format: NSLocalizedString("per_night_format", comment: ""), Double(pricePerNightCents) / 100.0)
    }
}

File: apps/android/app/src/main/java/com/sofaexchange/domain/model/Listing.kt
package com.sofaexchange.domain.model

data class Listing(
    val id: String,
    val title: String,
    val city: City,
    val pricePerNightCents: Int,
    val sofaType: SofaType,
    val hasFreeWifi: Boolean,
    val createdAt: String,
    val updatedAt: String,
)
